app/src/main/java/io/apicurio/registry/ccompat/rest/v7/impl/SubjectsResourceImpl.java

```diff
@@ -110,6 +111,12 @@ public Schema findSchemaByContent(String subject, SchemaInfo request, Boolean no
     public List<Integer> deleteSubject(String subject, Boolean permanent, String groupId) throws Exception {
         GA ga = getGA(groupId, subject);
 
+        List<Long> globalIdsReferencingArtifact = getStorage().getGlobalIdsReferencingArtifact(ga.getGroupId(), ga.getArtifactId());
+        if (!globalIdsReferencingArtifact.isEmpty()) {
+            // There are other schemas referencing this subject, it cannot be deleted.
+            throw new ReferenceExistsException(String.format("There are subjects referencing %s", subject));
```

Comment:
We could provide a list of subjects+versions / GAVs to the user so they know what causes the problem.